src/modules/cinema/cinema.schema.ts

```diff
@@ -0,0 +1,42 @@
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { CreateProjection, CreateMenuItem } from './dto/index';
+import mongoose, { HydratedDocument } from 'mongoose';
+import { Hall } from '../hall/hall.schema';
+
+export type CinemaDocument = HydratedDocument<Cinema>;
+
+@Schema()
+export class Cinema {
+  @Prop({ required: true, type: String })
+  address: string;
+
+  @Prop({ required: true, type: String })
+  name: string;
+
+  @Prop({ type: Number })
+  numberOfHalls: number;
+
+  @Prop({ required: true, type: Array })
+  menu: CreateMenuItem[];
+
+  @Prop({ required: true, type: Array })
+  projections: CreateProjection[];
```

Comment:
I think it's better to keep projections in a separate collection, and here you can either keep an array of projections IDs or remove this field since you will be able to get all the projections of the cinema by cinema's ID. 
The main reason for that is that user should be able to search projections across all the available projections of all the cinemas, it will be hard to implement if you keep the projections like that.